### PR TITLE
Do not update release when it has failed the max num of attempts

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#15213

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/giantswarm/appcatalog v0.3.2
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/exporterkit v0.2.0
-	github.com/giantswarm/helmclient/v4 v4.2.1
+	github.com/giantswarm/helmclient/v4 v4.2.2-0.20210128120700-4f0e7dd08e0c
```
Comment (rossf7) [inline go.mod:11]: TODO

**Review by rossf7 — COMMENTED** [review]
Tested on giraffe.

**Review by tomahawk28 — COMMENTED** [review]
I suggest changing the loop as below. 
```go
for i := 0; i < releaseFailedMaxAttempts; i++ {
  if history[i].Status != helmclient.StatusFailed {
    return false
  }
}

// if all failed, return true
return true
}
```

This is simpler, right?

File: service/controller/chart/resource/release/resource.go [hunk 2]
```diff
@@ -221,6 +227,36 @@ func (r *Resource) addHashAnnotation(ctx context.Context, cr v1alpha1.Chart, rel
...
+		return history[i].Revision > history[j].Revision
+	})
+
+	for _, hist := range history {
+		if hist.Status != helmclient.StatusFailed {
+			return false, nil
+		}
+
+		failCount++
+		if failCount >= releaseFailedMaxAttempts {
```
[the remark this answers is not in the document]
Comment (rossf7) [reply service/controller/chart/resource/release/resource.go:252]: Thanks suggestion looks good. I'll check it out fully when I get back.

[the remark this answers is not in the document]
Comment (rossf7) [reply service/controller/chart/resource/release/resource.go]: Yes that is much cleaner. I ran out of time today so I'll give this a final test on Monday.

**Review by tomahawk28 — APPROVED** [review]
LGTM!

**Pull request merged**